sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -344,16 +344,22 @@ private static string GetCanonicalName(string account, string fileSystemName, st
         /// </summary>
         private void EnsureState()
```

Comment:
Can we add unit tests that cover the various combinations of Identifier, Permissions, and ExpiresOn so we can make sure we are throwing when necessary?

Reply:
added